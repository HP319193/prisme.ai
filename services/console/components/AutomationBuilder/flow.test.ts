import { buildFlow } from './flow';

const automation = {
  name: 'example',
  trigger: {
    events: ['apps.foo', 'apps.bar'],
  },
  do: [
    {
      emit: {
        event: 'apps.foobar',
        payload: {
          foo: 'bar',
        },
        private: true,
      },
    },
    {
      wait: {
        event: 'apps.forIt',
        output: 'value',
      },
    },
    {
      set: {
        name: 'foo',
        value: '$value',
        lifespan: 42,
      },
    },
    {
      delete: {
        name: 'bar',
      },
    },
    {
      conditions: {
        '$a == 1': [
          {
            emit: 'apps.1',
          },
          {
            emit: 'apps.2',
          },
        ],
        default: [
          {
            all: [
              {
                emit: 'apps.3',
              },
              {
                emit: 'apps.4',
              },
              {
                repeat: {
                  on: '$a',
                  then: [
                    {
                      emit: 'apps.5',
                    },
                  ],
                },
              },
            ],
          },
        ],
      },
    },
  ],
};

it('should build flow', () => {
  expect(buildFlow(automation)).toMatchSnapshot();
});
