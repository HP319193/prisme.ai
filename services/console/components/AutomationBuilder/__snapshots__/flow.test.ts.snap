// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should build flow 1`] = `
Array [
  Object {
    "data": Object {
      "trigger": true,
      "value": undefined,
    },
    "id": "0",
    "position": Object {
      "x": 0,
      "y": 0,
    },
    "type": "trigger",
  },
  Object {
    "data": Object {
      "index": 0,
      "label": "emit",
      "parent": Array [
        Object {
          "emit": Object {
            "event": "apps.foobar",
            "payload": Object {
              "foo": "bar",
            },
            "private": true,
          },
        },
        Object {
          "wait": Object {
            "event": "apps.forIt",
            "output": "value",
          },
        },
        Object {
          "set": Object {
            "lifespan": 42,
            "name": "foo",
            "value": "$value",
          },
        },
        Object {
          "delete": Object {
            "name": "bar",
          },
        },
        Object {
          "conditions": Object {
            "$a == 1": Array [
              Object {
                "emit": "apps.1",
              },
              Object {
                "emit": "apps.2",
              },
            ],
            "default": Array [
              Object {
                "all": Array [
                  Object {
                    "emit": "apps.3",
                  },
                  Object {
                    "emit": "apps.4",
                  },
                  Object {
                    "repeat": Object {
                      "do": Array [
                        Object {
                          "emit": "apps.5",
                        },
                      ],
                      "on": "$a",
                    },
                  },
                ],
              },
            ],
          },
        },
      ],
      "value": Object {
        "event": "apps.foobar",
        "payload": Object {
          "foo": "bar",
        },
        "private": true,
      },
      "withButton": true,
    },
    "id": "0.0",
    "position": Object {
      "x": 0,
      "y": 150,
    },
    "type": "instruction",
  },
  Object {
    "data": Object {
      "index": 1,
      "label": "wait",
      "parent": Array [
        Object {
          "emit": Object {
            "event": "apps.foobar",
            "payload": Object {
              "foo": "bar",
            },
            "private": true,
          },
        },
        Object {
          "wait": Object {
            "event": "apps.forIt",
            "output": "value",
          },
        },
        Object {
          "set": Object {
            "lifespan": 42,
            "name": "foo",
            "value": "$value",
          },
        },
        Object {
          "delete": Object {
            "name": "bar",
          },
        },
        Object {
          "conditions": Object {
            "$a == 1": Array [
              Object {
                "emit": "apps.1",
              },
              Object {
                "emit": "apps.2",
              },
            ],
            "default": Array [
              Object {
                "all": Array [
                  Object {
                    "emit": "apps.3",
                  },
                  Object {
                    "emit": "apps.4",
                  },
                  Object {
                    "repeat": Object {
                      "do": Array [
                        Object {
                          "emit": "apps.5",
                        },
                      ],
                      "on": "$a",
                    },
                  },
                ],
              },
            ],
          },
        },
      ],
      "value": Object {
        "event": "apps.forIt",
        "output": "value",
      },
      "withButton": true,
    },
    "id": "0.1",
    "position": Object {
      "x": 0,
      "y": 300,
    },
    "type": "instruction",
  },
  Object {
    "data": Object {
      "index": 2,
      "label": "set",
      "parent": Array [
        Object {
          "emit": Object {
            "event": "apps.foobar",
            "payload": Object {
              "foo": "bar",
            },
            "private": true,
          },
        },
        Object {
          "wait": Object {
            "event": "apps.forIt",
            "output": "value",
          },
        },
        Object {
          "set": Object {
            "lifespan": 42,
            "name": "foo",
            "value": "$value",
          },
        },
        Object {
          "delete": Object {
            "name": "bar",
          },
        },
        Object {
          "conditions": Object {
            "$a == 1": Array [
              Object {
                "emit": "apps.1",
              },
              Object {
                "emit": "apps.2",
              },
            ],
            "default": Array [
              Object {
                "all": Array [
                  Object {
                    "emit": "apps.3",
                  },
                  Object {
                    "emit": "apps.4",
                  },
                  Object {
                    "repeat": Object {
                      "do": Array [
                        Object {
                          "emit": "apps.5",
                        },
                      ],
                      "on": "$a",
                    },
                  },
                ],
              },
            ],
          },
        },
      ],
      "value": Object {
        "lifespan": 42,
        "name": "foo",
        "value": "$value",
      },
      "withButton": true,
    },
    "id": "0.2",
    "position": Object {
      "x": 0,
      "y": 450,
    },
    "type": "instruction",
  },
  Object {
    "data": Object {
      "index": 3,
      "label": "delete",
      "parent": Array [
        Object {
          "emit": Object {
            "event": "apps.foobar",
            "payload": Object {
              "foo": "bar",
            },
            "private": true,
          },
        },
        Object {
          "wait": Object {
            "event": "apps.forIt",
            "output": "value",
          },
        },
        Object {
          "set": Object {
            "lifespan": 42,
            "name": "foo",
            "value": "$value",
          },
        },
        Object {
          "delete": Object {
            "name": "bar",
          },
        },
        Object {
          "conditions": Object {
            "$a == 1": Array [
              Object {
                "emit": "apps.1",
              },
              Object {
                "emit": "apps.2",
              },
            ],
            "default": Array [
              Object {
                "all": Array [
                  Object {
                    "emit": "apps.3",
                  },
                  Object {
                    "emit": "apps.4",
                  },
                  Object {
                    "repeat": Object {
                      "do": Array [
                        Object {
                          "emit": "apps.5",
                        },
                      ],
                      "on": "$a",
                    },
                  },
                ],
              },
            ],
          },
        },
      ],
      "value": Object {
        "name": "bar",
      },
      "withButton": true,
    },
    "id": "0.3",
    "position": Object {
      "x": 0,
      "y": 600,
    },
    "type": "instruction",
  },
  Object {
    "data": Object {
      "index": 4,
      "label": "conditions",
      "parent": Array [
        Object {
          "emit": Object {
            "event": "apps.foobar",
            "payload": Object {
              "foo": "bar",
            },
            "private": true,
          },
        },
        Object {
          "wait": Object {
            "event": "apps.forIt",
            "output": "value",
          },
        },
        Object {
          "set": Object {
            "lifespan": 42,
            "name": "foo",
            "value": "$value",
          },
        },
        Object {
          "delete": Object {
            "name": "bar",
          },
        },
        Object {
          "conditions": Object {
            "$a == 1": Array [
              Object {
                "emit": "apps.1",
              },
              Object {
                "emit": "apps.2",
              },
            ],
            "default": Array [
              Object {
                "all": Array [
                  Object {
                    "emit": "apps.3",
                  },
                  Object {
                    "emit": "apps.4",
                  },
                  Object {
                    "repeat": Object {
                      "do": Array [
                        Object {
                          "emit": "apps.5",
                        },
                      ],
                      "on": "$a",
                    },
                  },
                ],
              },
            ],
          },
        },
      ],
      "value": Object {
        "$a == 1": Array [
          Object {
            "emit": "apps.1",
          },
          Object {
            "emit": "apps.2",
          },
        ],
        "default": Array [
          Object {
            "all": Array [
              Object {
                "emit": "apps.3",
              },
              Object {
                "emit": "apps.4",
              },
              Object {
                "repeat": Object {
                  "do": Array [
                    Object {
                      "emit": "apps.5",
                    },
                  ],
                  "on": "$a",
                },
              },
            ],
          },
        ],
      },
      "withButton": true,
    },
    "id": "0.4",
    "position": Object {
      "x": 0,
      "y": 750,
    },
    "type": "conditions",
  },
  Object {
    "data": Object {
      "index": 0,
      "label": "emit",
      "parent": Array [
        Object {
          "emit": "apps.1",
        },
        Object {
          "emit": "apps.2",
        },
      ],
      "value": "apps.1",
      "withButton": true,
    },
    "id": "0.4.0.0",
    "position": Object {
      "x": 300,
      "y": 900,
    },
    "type": "instruction",
  },
  Object {
    "data": Object {
      "index": 1,
      "label": "emit",
      "parent": Array [
        Object {
          "emit": "apps.1",
        },
        Object {
          "emit": "apps.2",
        },
      ],
      "value": "apps.2",
      "withButton": true,
    },
    "id": "0.4.0.1",
    "position": Object {
      "x": 300,
      "y": 1050,
    },
    "type": "instruction",
  },
  Object {
    "data": Object {
      "index": 2,
      "label": "empty block",
      "parent": Array [
        Object {
          "emit": "apps.1",
        },
        Object {
          "emit": "apps.2",
        },
      ],
      "value": null,
      "withButton": true,
    },
    "id": "0.4.0.2",
    "position": Object {
      "x": 300,
      "y": 1200,
    },
    "type": "empty",
  },
  Object {
    "data": Object {
      "index": 0,
      "label": "all",
      "parent": Array [
        Object {
          "all": Array [
            Object {
              "emit": "apps.3",
            },
            Object {
              "emit": "apps.4",
            },
            Object {
              "repeat": Object {
                "do": Array [
                  Object {
                    "emit": "apps.5",
                  },
                ],
                "on": "$a",
              },
            },
          ],
        },
      ],
      "value": Array [
        Object {
          "emit": "apps.3",
        },
        Object {
          "emit": "apps.4",
        },
        Object {
          "repeat": Object {
            "do": Array [
              Object {
                "emit": "apps.5",
              },
            ],
            "on": "$a",
          },
        },
      ],
      "withButton": true,
    },
    "id": "0.4.1.0",
    "position": Object {
      "x": 600,
      "y": 900,
    },
    "type": "instruction",
  },
  Object {
    "data": Object {
      "index": 0,
      "label": "emit",
      "parent": Array [
        Object {
          "emit": "apps.3",
        },
        Object {
          "emit": "apps.4",
        },
        Object {
          "repeat": Object {
            "do": Array [
              Object {
                "emit": "apps.5",
              },
            ],
            "on": "$a",
          },
        },
      ],
      "value": Array [
        Object {
          "emit": "apps.3",
        },
        Object {
          "emit": "apps.4",
        },
        Object {
          "repeat": Object {
            "do": Array [
              Object {
                "emit": "apps.5",
              },
            ],
            "on": "$a",
          },
        },
      ],
      "withButton": true,
    },
    "id": "0.4.1.0.0",
    "position": Object {
      "x": 900,
      "y": 1050,
    },
    "type": "instruction",
  },
  Object {
    "data": Object {
      "index": 1,
      "label": "emit",
      "parent": Array [
        Object {
          "emit": "apps.3",
        },
        Object {
          "emit": "apps.4",
        },
        Object {
          "repeat": Object {
            "do": Array [
              Object {
                "emit": "apps.5",
              },
            ],
            "on": "$a",
          },
        },
      ],
      "value": Array [
        Object {
          "emit": "apps.3",
        },
        Object {
          "emit": "apps.4",
        },
        Object {
          "repeat": Object {
            "do": Array [
              Object {
                "emit": "apps.5",
              },
            ],
            "on": "$a",
          },
        },
      ],
      "withButton": true,
    },
    "id": "0.4.1.0.1",
    "position": Object {
      "x": 1200,
      "y": 1050,
    },
    "type": "instruction",
  },
  Object {
    "data": Object {
      "index": 2,
      "label": "repeat",
      "parent": Array [
        Object {
          "emit": "apps.3",
        },
        Object {
          "emit": "apps.4",
        },
        Object {
          "repeat": Object {
            "do": Array [
              Object {
                "emit": "apps.5",
              },
            ],
            "on": "$a",
          },
        },
      ],
      "value": Array [
        Object {
          "emit": "apps.3",
        },
        Object {
          "emit": "apps.4",
        },
        Object {
          "repeat": Object {
            "do": Array [
              Object {
                "emit": "apps.5",
              },
            ],
            "on": "$a",
          },
        },
      ],
      "withButton": true,
    },
    "id": "0.4.1.0.2",
    "position": Object {
      "x": 1500,
      "y": 1050,
    },
    "type": "instruction",
  },
  Object {
    "data": Object {
      "index": 3,
      "label": "new all",
      "parent": Array [
        Object {
          "emit": "apps.3",
        },
        Object {
          "emit": "apps.4",
        },
        Object {
          "repeat": Object {
            "do": Array [
              Object {
                "emit": "apps.5",
              },
            ],
            "on": "$a",
          },
        },
      ],
      "value": Array [
        Object {
          "emit": "apps.3",
        },
        Object {
          "emit": "apps.4",
        },
        Object {
          "repeat": Object {
            "do": Array [
              Object {
                "emit": "apps.5",
              },
            ],
            "on": "$a",
          },
        },
      ],
      "withButton": true,
    },
    "id": "0.4.1.0.3",
    "position": Object {
      "x": 1800,
      "y": 1050,
    },
    "type": "empty",
  },
  Object {
    "data": Object {
      "index": 1,
      "label": "empty block",
      "parent": Array [
        Object {
          "all": Array [
            Object {
              "emit": "apps.3",
            },
            Object {
              "emit": "apps.4",
            },
            Object {
              "repeat": Object {
                "do": Array [
                  Object {
                    "emit": "apps.5",
                  },
                ],
                "on": "$a",
              },
            },
          ],
        },
      ],
      "value": null,
      "withButton": true,
    },
    "id": "0.4.1.1",
    "position": Object {
      "x": 600,
      "y": 1200,
    },
    "type": "empty",
  },
  Object {
    "data": Object {
      "key": "new condition",
      "parent": Object {
        "conditions": Object {
          "$a == 1": Array [
            Object {
              "emit": "apps.1",
            },
            Object {
              "emit": "apps.2",
            },
          ],
          "default": Array [
            Object {
              "all": Array [
                Object {
                  "emit": "apps.3",
                },
                Object {
                  "emit": "apps.4",
                },
                Object {
                  "repeat": Object {
                    "do": Array [
                      Object {
                        "emit": "apps.5",
                      },
                    ],
                    "on": "$a",
                  },
                },
              ],
            },
          ],
        },
      },
      "withButton": false,
    },
    "id": "0.4.2.0",
    "position": Object {
      "x": 900,
      "y": 900,
    },
    "type": "empty",
  },
  Object {
    "data": Object {
      "index": 5,
      "label": "empty block",
      "parent": Array [
        Object {
          "emit": Object {
            "event": "apps.foobar",
            "payload": Object {
              "foo": "bar",
            },
            "private": true,
          },
        },
        Object {
          "wait": Object {
            "event": "apps.forIt",
            "output": "value",
          },
        },
        Object {
          "set": Object {
            "lifespan": 42,
            "name": "foo",
            "value": "$value",
          },
        },
        Object {
          "delete": Object {
            "name": "bar",
          },
        },
        Object {
          "conditions": Object {
            "$a == 1": Array [
              Object {
                "emit": "apps.1",
              },
              Object {
                "emit": "apps.2",
              },
            ],
            "default": Array [
              Object {
                "all": Array [
                  Object {
                    "emit": "apps.3",
                  },
                  Object {
                    "emit": "apps.4",
                  },
                  Object {
                    "repeat": Object {
                      "do": Array [
                        Object {
                          "emit": "apps.5",
                        },
                      ],
                      "on": "$a",
                    },
                  },
                ],
              },
            ],
          },
        },
      ],
      "value": null,
      "withButton": true,
    },
    "id": "0.5",
    "position": Object {
      "x": 0,
      "y": 1350,
    },
    "type": "empty",
  },
  Object {
    "arrowHeadType": "arrow",
    "data": Object {
      "index": 1,
      "parent": Array [
        Object {
          "emit": Object {
            "event": "apps.foobar",
            "payload": Object {
              "foo": "bar",
            },
            "private": true,
          },
        },
        Object {
          "wait": Object {
            "event": "apps.forIt",
            "output": "value",
          },
        },
        Object {
          "set": Object {
            "lifespan": 42,
            "name": "foo",
            "value": "$value",
          },
        },
        Object {
          "delete": Object {
            "name": "bar",
          },
        },
        Object {
          "conditions": Object {
            "$a == 1": Array [
              Object {
                "emit": "apps.1",
              },
              Object {
                "emit": "apps.2",
              },
            ],
            "default": Array [
              Object {
                "all": Array [
                  Object {
                    "emit": "apps.3",
                  },
                  Object {
                    "emit": "apps.4",
                  },
                  Object {
                    "repeat": Object {
                      "do": Array [
                        Object {
                          "emit": "apps.5",
                        },
                      ],
                      "on": "$a",
                    },
                  },
                ],
              },
            ],
          },
        },
      ],
    },
    "id": "edge-0.0-0.1",
    "source": "0.0",
    "target": "0.1",
    "type": "instruction",
  },
  Object {
    "arrowHeadType": "arrow",
    "data": Object {
      "index": 2,
      "parent": Array [
        Object {
          "emit": Object {
            "event": "apps.foobar",
            "payload": Object {
              "foo": "bar",
            },
            "private": true,
          },
        },
        Object {
          "wait": Object {
            "event": "apps.forIt",
            "output": "value",
          },
        },
        Object {
          "set": Object {
            "lifespan": 42,
            "name": "foo",
            "value": "$value",
          },
        },
        Object {
          "delete": Object {
            "name": "bar",
          },
        },
        Object {
          "conditions": Object {
            "$a == 1": Array [
              Object {
                "emit": "apps.1",
              },
              Object {
                "emit": "apps.2",
              },
            ],
            "default": Array [
              Object {
                "all": Array [
                  Object {
                    "emit": "apps.3",
                  },
                  Object {
                    "emit": "apps.4",
                  },
                  Object {
                    "repeat": Object {
                      "do": Array [
                        Object {
                          "emit": "apps.5",
                        },
                      ],
                      "on": "$a",
                    },
                  },
                ],
              },
            ],
          },
        },
      ],
    },
    "id": "edge-0.1-0.2",
    "source": "0.1",
    "target": "0.2",
    "type": "instruction",
  },
  Object {
    "arrowHeadType": "arrow",
    "data": Object {
      "index": 3,
      "parent": Array [
        Object {
          "emit": Object {
            "event": "apps.foobar",
            "payload": Object {
              "foo": "bar",
            },
            "private": true,
          },
        },
        Object {
          "wait": Object {
            "event": "apps.forIt",
            "output": "value",
          },
        },
        Object {
          "set": Object {
            "lifespan": 42,
            "name": "foo",
            "value": "$value",
          },
        },
        Object {
          "delete": Object {
            "name": "bar",
          },
        },
        Object {
          "conditions": Object {
            "$a == 1": Array [
              Object {
                "emit": "apps.1",
              },
              Object {
                "emit": "apps.2",
              },
            ],
            "default": Array [
              Object {
                "all": Array [
                  Object {
                    "emit": "apps.3",
                  },
                  Object {
                    "emit": "apps.4",
                  },
                  Object {
                    "repeat": Object {
                      "do": Array [
                        Object {
                          "emit": "apps.5",
                        },
                      ],
                      "on": "$a",
                    },
                  },
                ],
              },
            ],
          },
        },
      ],
    },
    "id": "edge-0.2-0.3",
    "source": "0.2",
    "target": "0.3",
    "type": "instruction",
  },
  Object {
    "arrowHeadType": "arrow",
    "data": Object {
      "index": 4,
      "parent": Array [
        Object {
          "emit": Object {
            "event": "apps.foobar",
            "payload": Object {
              "foo": "bar",
            },
            "private": true,
          },
        },
        Object {
          "wait": Object {
            "event": "apps.forIt",
            "output": "value",
          },
        },
        Object {
          "set": Object {
            "lifespan": 42,
            "name": "foo",
            "value": "$value",
          },
        },
        Object {
          "delete": Object {
            "name": "bar",
          },
        },
        Object {
          "conditions": Object {
            "$a == 1": Array [
              Object {
                "emit": "apps.1",
              },
              Object {
                "emit": "apps.2",
              },
            ],
            "default": Array [
              Object {
                "all": Array [
                  Object {
                    "emit": "apps.3",
                  },
                  Object {
                    "emit": "apps.4",
                  },
                  Object {
                    "repeat": Object {
                      "do": Array [
                        Object {
                          "emit": "apps.5",
                        },
                      ],
                      "on": "$a",
                    },
                  },
                ],
              },
            ],
          },
        },
      ],
    },
    "id": "edge-0.3-0.4",
    "source": "0.3",
    "target": "0.4",
    "type": "instruction",
  },
  Object {
    "arrowHeadType": "arrow",
    "data": Object {
      "index": 1,
      "parent": Array [
        Object {
          "emit": "apps.1",
        },
        Object {
          "emit": "apps.2",
        },
      ],
    },
    "id": "edge-0.4.0.0-0.4.0.1",
    "source": "0.4.0.0",
    "target": "0.4.0.1",
    "type": "instruction",
  },
  Object {
    "arrowHeadType": "arrow",
    "data": Object {
      "index": 2,
      "parent": Array [
        Object {
          "emit": "apps.1",
        },
        Object {
          "emit": "apps.2",
        },
      ],
    },
    "id": "edge-0.4.0.1-0.4.0.2",
    "source": "0.4.0.1",
    "target": "0.4.0.2",
    "type": "edge",
  },
  Object {
    "arrowHeadType": "arrow",
    "data": Object {
      "key": "$a == 1",
      "label": "$a == 1",
      "parent": Array [
        Object {
          "emit": "apps.1",
        },
        Object {
          "emit": "apps.2",
        },
      ],
    },
    "id": "edge-0.4-0.4.0.0",
    "source": "0.4",
    "sourceHandle": "$a == 1",
    "target": "0.4.0.0",
    "type": "conditionEdge",
  },
  Object {
    "arrowHeadType": "arrow",
    "id": "edge-0.4.1.0-0.4.1.0.0",
    "source": "0.4.1.0",
    "target": "0.4.1.0.0",
    "type": "edge",
  },
  Object {
    "arrowHeadType": "arrow",
    "id": "edge-0.4.1.0.0-0.4.1.1",
    "source": "0.4.1.0.0",
    "target": "0.4.1.1",
    "type": "edge",
  },
  Object {
    "arrowHeadType": "arrow",
    "id": "edge-0.4.1.0-0.4.1.0.1",
    "source": "0.4.1.0",
    "target": "0.4.1.0.1",
    "type": "edge",
  },
  Object {
    "arrowHeadType": "arrow",
    "id": "edge-0.4.1.0.1-0.4.1.1",
    "source": "0.4.1.0.1",
    "target": "0.4.1.1",
    "type": "edge",
  },
  Object {
    "arrowHeadType": "arrow",
    "id": "edge-0.4.1.0-0.4.1.0.2",
    "source": "0.4.1.0",
    "target": "0.4.1.0.2",
    "type": "edge",
  },
  Object {
    "arrowHeadType": "arrow",
    "id": "edge-0.4.1.0.2-0.4.1.1",
    "source": "0.4.1.0.2",
    "target": "0.4.1.1",
    "type": "edge",
  },
  Object {
    "arrowHeadType": "arrow",
    "id": "edge-0.4.1.0-0.4.1.0.3",
    "source": "0.4.1.0",
    "target": "0.4.1.0.3",
    "type": "edge",
  },
  Object {
    "arrowHeadType": "arrow",
    "id": "edge-0.4.1.0.3-0.4.1.1",
    "source": "0.4.1.0.3",
    "target": "0.4.1.1",
    "type": "edge",
  },
  Object {
    "arrowHeadType": "arrow",
    "data": Object {
      "key": "default",
      "label": "default",
      "parent": undefined,
    },
    "id": "edge-0.4-0.4.1.0",
    "source": "0.4",
    "sourceHandle": "default",
    "target": "0.4.1.0",
    "type": "conditionEdge",
  },
  Object {
    "arrowHeadType": "arrow",
    "data": Object {
      "key": "",
      "label": "new condition",
      "parent": Object {
        "conditions": Object {
          "$a == 1": Array [
            Object {
              "emit": "apps.1",
            },
            Object {
              "emit": "apps.2",
            },
          ],
          "default": Array [
            Object {
              "all": Array [
                Object {
                  "emit": "apps.3",
                },
                Object {
                  "emit": "apps.4",
                },
                Object {
                  "repeat": Object {
                    "do": Array [
                      Object {
                        "emit": "apps.5",
                      },
                    ],
                    "on": "$a",
                  },
                },
              ],
            },
          ],
        },
      },
    },
    "id": "edge-0.4-0.4.2.0",
    "source": "0.4",
    "sourceHandle": "new condition",
    "target": "0.4.2.0",
    "type": "conditionEdge",
  },
  Object {
    "arrowHeadType": "arrow",
    "id": "edge-0.5-0.4.0.2",
    "source": "0.4.0.2",
    "target": "0.5",
    "type": "conditionEdge",
  },
  Object {
    "arrowHeadType": "arrow",
    "id": "edge-0.5-0.4.1.1",
    "source": "0.4.1.1",
    "target": "0.5",
    "type": "conditionEdge",
  },
  Object {
    "arrowHeadType": "arrow",
    "id": "edge-0.5-0.4.2.0",
    "source": "0.4.2.0",
    "target": "0.5",
    "type": "conditionEdge",
  },
  Object {
    "arrowHeadType": "arrow",
    "data": Object {
      "index": 0,
      "parent": Array [
        Object {
          "emit": Object {
            "event": "apps.foobar",
            "payload": Object {
              "foo": "bar",
            },
            "private": true,
          },
        },
        Object {
          "wait": Object {
            "event": "apps.forIt",
            "output": "value",
          },
        },
        Object {
          "set": Object {
            "lifespan": 42,
            "name": "foo",
            "value": "$value",
          },
        },
        Object {
          "delete": Object {
            "name": "bar",
          },
        },
        Object {
          "conditions": Object {
            "$a == 1": Array [
              Object {
                "emit": "apps.1",
              },
              Object {
                "emit": "apps.2",
              },
            ],
            "default": Array [
              Object {
                "all": Array [
                  Object {
                    "emit": "apps.3",
                  },
                  Object {
                    "emit": "apps.4",
                  },
                  Object {
                    "repeat": Object {
                      "do": Array [
                        Object {
                          "emit": "apps.5",
                        },
                      ],
                      "on": "$a",
                    },
                  },
                ],
              },
            ],
          },
        },
      ],
    },
    "id": "edge-0-0.0",
    "source": "0",
    "target": "0.0",
    "type": "instruction",
  },
]
`;
